final_project/flaskr/templates/scripts.js

```diff
@@ -358,28 +358,25 @@ BikeMap.prototype.addTempMarker = function(lat, lng, userId, address) {
 
 BikeMap.prototype.createMarker = function(state) {
     
```

Comment:
I think what you want to do here is have a variable `marker` that you declare up front. This will be what gets returned from `createMarker`.

Loop over the layers and if you find a matching one, set the value of `marker` to that layer.
If, after the loop, `marker` still isn't set, then make a new one and the value of `marker` to that.

Then, update `marker` with whatever values you expect it to have. If the marker already existed, some of this will probably be redundant. That's OK.

Reply:
👍